conans/client/conan_api.py

```diff
@@ -589,13 +590,30 @@ def copy(self, reference, user_channel, force=False, packages=None):
                  self._user_io, self._remote_manager, force=force)
 
     @api_method
-    def user(self, name=None, clean=False, remote=None, password=None):
-        if clean:
-            localdb = LocalDB(self._client_cache.localdb)
-            localdb.init(clean=True)
-            self._user_io.out.success("Deleted user data")
-            return
-        self._manager.user(remote, name, password)
+    def authenticate(self, name, password, remote=None):
+        registry = RemoteRegistry(self._client_cache.registry, self._user_io.out)
+        if not remote:
+            remote = registry.default_remote
+        else:
+            remote = registry.remote(remote)
+        if password == "":
+            name, password = self._user_io.request_login(remote_name=remote.name,
```

Comment:
I think the python API should never request an input. Extract it to command.

Reply:
Also, it would be very ugly to have the following code in ``command.py``:

```python
registry = RemoteRegistry(self._client_cache.registry, self._user_io.out)
        if not remote:
            remote = registry.default_remote
        else:
            remote = registry.remote(remote)
```

Still think it is necessary to move it out the conan_api?